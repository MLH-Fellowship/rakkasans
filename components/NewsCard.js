import React from "react";
import { View, Text, StyleSheet, Image, TouchableOpacity } from "react-native";

import Colors from "../constants/Colors";
import Dimensions from "../constants/Dimensions";

export default function NewsCard({ image, title, onPress }) {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={onPress}
      activeOpacity={0.4}
    >
      <Image
        style={{ width: "100%", height: "100%" }}
        resizeMode="cover"
        source={image}
      />
      <Text style={[styles.titleText, Dimensions.font]}>{title}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.darkGray,
    marginTop: 25,
    width: "100%",
    height: 175,
    alignItems: "center",
    borderWidth: 1,
    borderColor: Colors.primary,
  },
  titleText: {
    backgroundColor: Colors.primary,
    padding: 8,
    paddingBottom: 2,
    borderColor: Colors.white,
    borderWidth: 1,
    color: Colors.white,
    fontSize: 16,
    position: "absolute",
    bottom: 15,
    left: 15,
    shadowOffset: { width: 5, height: 5 },
    shadowColor: "black",
    shadowOpacity: 1.0,
  },
});
