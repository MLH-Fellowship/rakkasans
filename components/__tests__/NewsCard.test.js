import React from 'react';
import renderer from 'react-test-renderer';

import NewsCard from '../NewsCard';

jest.useFakeTimers();

it('renders correctly', () => {
  const tree = renderer.create(<NewsCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
