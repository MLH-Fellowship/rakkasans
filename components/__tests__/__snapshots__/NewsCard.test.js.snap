// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#333333",
      "height": 175,
      "marginTop": 25,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "#020a41",
          "borderColor": "#ffffff",
          "borderWidth": 1,
          "bottom": 15,
          "color": "#ffffff",
          "fontSize": 16,
          "left": 15,
          "padding": 8,
          "paddingBottom": 2,
          "position": "absolute",
        },
        Object {
          "fontFamily": "andale-mono",
        },
      ]
    }
  />
</View>
`;
